### ISSUE TITLE
phpbrew broken after self-update failure

### ISSUE THREAD

**Issue by jhdxr** [phpbrew/phpbrew#524]
here is the log. It seems although curl get nothing from server, phpbrew still overwrite the empty string to the executable phar

env: osX 10.10
```
xiami@xiamideMacBook-Pro ~/tmp> sudo phpbrew self-update
Password:
Updating phpbrew /usr/bin/phpbrew from master...

curl: (52) Empty reply from server
Version updated.
xiami@xiamideMacBook-Pro ~/tmp> phpbrew -v
Failed to execute process '/usr/bin/phpbrew'. Reason:
exec: Exec format error
The file '/usr/bin/phpbrew' is marked as an executable but could not be run by the operating system.
xiami@xiamideMacBook-Pro /tmp> ll /usr/bin/phpbrew 
-rwxr-xr-x  1 xiami  wheel     0B  6 15 11:30 /usr/bin/phpbrew
```

**Comment by marcioAlmada** [comment]
Can you `cat /usr/bin/phpbrew`? I just tested `self-update` and it seems fine. We could download to a tmp file and only overwrite if successful to avoid this kind of error, you probably had a bad transfer.

**Comment by jhdxr** [comment]
@marcioAlmada I've reinstalled phpbrew, but you can read last line of my log, it indicates /usr/bin/phpbrew is an empty file(0B). 
```
xiami@xiamideMacBook-Pro /tmp> ll /usr/bin/phpbrew 
-rwxr-xr-x  1 xiami  wheel     0B  6 15 11:30 /usr/bin/phpbrew
```

**Comment by jhdxr** [comment]
there's an easy way to reproduce this problem, use `phpbrew self-update`, and press ctrl+c when downloading.

**Comment by c9s** [comment]
I think you need to re-download the phar file from master.

**Comment by jhdxr** [comment]
@c9s yes, of course I can, and it's how I fix the phpbrew. However I think the program should have the ability to detect exceptions and process properly.

**Comment by marcioAlmada** [comment]
fixed with #525
